core/client/src/main/java/alluxio/client/file/FileOutStream.java

```diff
@@ -116,7 +119,7 @@ public FileOutStream(AlluxioURI path, OutStreamOptions options) throws IOExcepti
         String tmpPath = PathUtils.temporaryFileName(mNonce, mUfsPath);
         UnderFileSystem ufs = UnderFileSystem.get(tmpPath, ClientContext.getConf());
         // TODO(jiri): Implement collection of temporary files left behind by dead clients.
-        mUnderStorageOutputStream = ufs.create(tmpPath, (int) mBlockSize);
+        mUnderStorageOutputStream = ufs.create(tmpPath, options.getPermissionStatus());
```

Comment:
Corrected with using `UnderFileSystemCreateOptions`.
Previously because the `BlockSize` is not actually consumed in UFS `create()`. See the `HdfsUnderFileSystem` implementation for details.